### DISCUSSION TITLE
Suggestion to lock the mouse cursor in the y axis when osu catch has the relax mod enabled

### DISCUSSION THREAD

**Discussion by RoyceThomas** [ppy/osu#27510] [Ideas] [answered]
Suggestion to lock the mouse cursor to the y axis when osu catch has the relax mod enabled as sometimes the mouse goes to the edges of the screen, and then it affects the movement of the catcher as the movements of the physical mouse don't reflect what is shown on the screen. My suggestion is to lock the mouse curser in the y axis so only x axis movements of the mouse is translated to osu.
Feel free to let me know if this causes any problems.

**Comment by Rekunan** [comment]
> movements of the physical mouse don't reflect what is shown on the screen

This sounds like an issue that should be fixed rather than creating an option just for this.

Can you send a video of this? I've failing to recreate it on my end.

**Reply by RoyceThomas** [reply]
It might be hard to show this in a video because the cursor is invisible during gameplay, and I don’t know how to show my physical mouse movements at the same time as well, but I’ll try explain it as best i can.
So during gameplay of osu catch with the relax mod on, after some gameplay, I find my cursor moving towards either the top or bottom edges of the screen. 

So let’s say at a given point in time during gameplay, the cursor is at the top edge of the screen and I move the physical mouse to the right, but also move the mouse a bit up with that movement (so essentially i mouse the mouse up right diagonally). Since the cursor is already fixed on the top of the screen, for some reason, I believe only some of the movements are recorded in our laser and thus, the actual platter moves right a bit less that what it did in real life.

Now you might be thinking theoretically if I did move the mouse up and right diagonally, even with this tweak that I’m suggesting, it won’t make a difference as osu! should only care about X axis movements  but, I still think it might make a difference. If the mouses X movements are the only ones recorded all the time, our brains would adapt to play with the new controls, and I believe this would be slightly more accurate. Additionally, if the cursor doesn’t drift to the top or bottom of the screen, the movements that I make with the mouse that are in the X axis would be counted all the time  instead of some X axis movements not being counted fully after i have reached the top or bottom edge or corners of the screen. 

Again, it’s hard to show this in a screen capture, and I’m not sure if I’ve explained it well. Maybe if you want to try and simulate yourself practice a level for about 30 seconds with the mouse at the centre of the screen playing normally and then halfway through just move your mouse to the top or bottom of the screen and play normally again, and you might feel a difference? Essentially, maybe having a toggle then of the cursor being locked in the midline of the screen in the Y axis, so that the cursor only moves left and right in that mode would fix the problem. 

PS as a side note, funnily enough I’ve experienced this more in osu laser than in the standard application. Not sure why maybe there’s a different y axis mouse acceleration…idk.
also again, not sure if this makes any difference, but I play in fullscreen with the cursor locked in the window at all times.

**Reply by Rekunan** [reply, marked as answer]
I keep trying for longer, even comparing to how stable feels, and I'm still not experiencing your issue.

Your issue also sounds really jank because the x movements are the only ones that move the catcher, any y movements shouldn't be moving the catcher at all, so it shouldn't matter in the first place if ur mouse is at the vertical edges or not.

**Reply by RoyceThomas** [reply]
actually i also don't experience it anymore as well. sorry! disregard this suggestion, but i am certain this was occurring before, but maybe due to my actual mouse idk, but guess its not there anymore... im not crazy!! 😣
edit: i still feel like at some rare occurrences its happening to me at least....  maybe having an option to see the cursor during gameplay can help diagnose the issue?
edit 2: actually nvm... i think even that wont solve the issue anyway.. ill try do my own troubleshooting and see if i find anything.
